File: tests/fixtures/csharp/deep_nesting.cs
public class Processor
{
    public void DeeplyNested(object[] data)
    {
        foreach (var group in data)
        {
            if (group != null)
            {
                foreach (var item in new object[] { group })
                {
                    if (item != null)
                    {
                        foreach (var sub in new object[] { item })
                        {
                            if (sub != null)
                            {
                                Process(sub);
                            }
                        }
                    }
                }
            }
        }
    }

    public void ModeratelyNested(object[] data)
    {
        foreach (var item in data)
        {
            if (item != null)
            {
                foreach (var sub in new object[] { item })
                {
                    Process(sub);
                }
            }
        }
    }

    private void Process(object x) { }
}

File: tests/fixtures/kotlin/DeepNesting.kt
class DeepNesting {
    fun deeplyNested(data: List<List<List<Int>>>): Boolean {
        for (group in data) {
            if (group.isNotEmpty()) {
                for (items in group) {
                    if (items.isNotEmpty()) {
                        for (v in items) {
                            if (v > 0) {
                                println(v)
                            }
                        }
                    }
                }
            }
        }
        return true
    }

    fun moderatelyNested(data: List<Int>): Int {
        var sum = 0
        for (item in data) {
            if (item > 0) {
                sum += item
            }
        }
        return sum
    }
}
